### ISSUE TITLE
Why not jiffy?

### ISSUE THREAD

**Issue by si14** [sockjs/sockjs-erlang#7]
Guys, we are living in 21 century, why don't you use jiffy instead of mochijson?

**Comment by simonmacmullen** [comment]
I dunno, ignorance? For me it doesn't show up in the first 10 pages searching for "erlang json" so it might as well not exist ;-)

Sarcasm aside, we already support eep0018 so jiffy should not be a stretch.

**Comment by majek** [comment]
(FTR: jiffy lives here https://github.com/davisp/jiffy)

Also: [...] premature optimization [...]

I think we should make SockJS stable first and than play with not-pure-erlang dependencies.

Additionally, there are high chances that JSON encoding will not be used for SockJS-protocol v0.2.

**Comment by simonmacmullen** [comment]
(Of course we can revisit this if the new protocol still needs JSON...)

**Comment by majek** [comment]
In #9 we came back to the idea.

**Comment by si14** [comment]
Nice to hear.

**Comment by gleber** [comment]
It seems that it won't be a premature optimization. Here's the output of fprof analysis over application which does broadcast data to ~3000 users 4 times (short messages around 110-120 bytes each, excluding SockJS protocol overhead):

https://gist.github.com/2781872

As you can see mochijson2 and it's utf8 handling code is the most CPU hungry one.

**Comment by si14** [comment]
http://si14.livejournal.com/18757.html you can also find some JSON-related benchmarks here.

**Comment by majek** [comment]
@gleber Jiffy is fine. We actually did support it in previous sockjs-erlang versions:
- https://github.com/sockjs/sockjs-erlang/blob/fb32ebba863770423c32a9b8e70dd458e75d0c79/src/sockjs_util.erl#L36

But, SockJS requires custom encoding of some unicode characters. Mochiweb does most of the work out of the box and to get everything working (`\uFFFE and \uFFFF` chars) I needed to patch it a bit.

I'm happy to have optional `jiffy` support if it passes all the sockjs-protocol tests regarding encoding:
- http://sockjs.github.com/sockjs-protocol/sockjs-protocol-0.3.html#section-129
